tests/foreman/ui_airgun/test_repository.py

```diff
@@ -131,6 +140,122 @@ def test_positive_create_as_non_admin_user(module_org, test_name):
             product.name, repo_name)[0]['Name'] == repo_name
 
 
+@tier2
+@upgrade
+def test_positive_create_puppet_repo_same_url_different_orgs(
+        session, module_prod):
+    """Create two repos with the same URL in two different organizations.
+
+    :id: f4cb00ed-6faf-4c79-9f66-76cd333299cb
+
+    :expectedresults: Repositories are created and puppet modules are visible
+        from different organizations.
+
+    :CaseLevel: Integration
+    """
+    # Create first repository
+    repo = entities.Repository(
+        url=FAKE_8_PUPPET_REPO,
+        product=module_prod,
+        content_type=REPO_TYPE['puppet'],
+    ).create()
+    repo.sync()
+    # Create second repository
+    org = entities.Organization().create()
+    product = entities.Product(organization=org).create()
+    new_repo = entities.Repository(
+        url=FAKE_8_PUPPET_REPO,
+        product=product,
+        content_type=REPO_TYPE['puppet'],
+    ).create()
+    new_repo.sync()
+    with session:
+        # Check packages number in first repository
+        assert session.repository.search(
+            module_prod.name, repo.name)[0]['Name'] == repo.name
+        repo = session.repository.read(module_prod.name, repo.name)
+        assert repo['content_counts']['Puppet Modules'] == '1'
+        # Check packages number in first repository
+        session.organization.select(org.name)
+        assert session.repository.search(
+            product.name, new_repo.name)[0]['Name'] == new_repo.name
+        new_repo = session.repository.read(product.name, new_repo.name)
+        assert new_repo['content_counts']['Puppet Modules'] == '1'
+
+
+@tier2
+@upgrade
+def test_positive_create_as_non_admin_user_with_cv_published(
+        module_org, module_prod, test_name):
+    """Create a repository as a non admin user in a product that already
+    contain a repository that is used in a published content view.
+
+    :id: 407864eb-50b8-4bc8-bbc7-0e6f8136d89f
+
+    :expectedresults: New repository successfully created by non admin user
+
+    :BZ: 1447829
+
+    :CaseLevel: Integration
+    """
+    user_login = gen_string('alpha')
+    user_password = gen_string('alphanumeric')
+    repo_name = gen_string('alpha')
+    user_permissions = {
+        None: ['access_dashboard'],
+        'Katello::Product': [
+            'view_products',
+            'create_products',
+            'edit_products',
+            'destroy_products',
+            'sync_products',
+            'export_products',
+        ],
+    }
+    role = entities.Role().create()
+    create_role_permissions(role, user_permissions)
+    entities.User(
+        login=user_login,
+        password=user_password,
+        role=[role],
+        admin=False,
+        default_organization=module_org,
+        organization=[module_org],
+    ).create()
+    repo = entities.Repository(
+        product=module_prod, url=FAKE_2_YUM_REPO).create()
```

Comment:
please isolate that step and use new product here. I want to be sure that product is empty when you start the flow